src/Services/TimeManager.vala

```diff
@@ -40,14 +43,22 @@ public class DateTime.Services.TimeManager : Gtk.Calendar {
 
         add_timeout ();
         try {
+            clockSettings = new GLib.Settings ("org.gnome.desktop.interface");
```

Comment:
You shouldn't probably create the same settings twice: here and in `PanelLabel`. You should setup a singleton for these settings so you can call like `get_clock_settings` and it would return you always the same instance.